keanu-project/src/main/java/io/improbable/keanu/templating/SequenceBuilder.java

```diff
@@ -240,6 +252,18 @@ private VertexLabel getDefaultParentLabel(VertexLabel proxyLabel) {
         }
     }
 
+    /**
+     * This function is best described by how it operates on labels passed to it:
+     *  1. `Sequence_Item_TIMESTEP.HASHCODE.proxy_for.LABEL` -> `proxy_for.LABEL`
+     *  2. `IDENTIFYING_NAMESPACE.Sequence_Item_TIMESTEP.HASHCODE.proxy_for.LABEL` ->  `proxy_for.LABEL`
+     */
+    private VertexLabel getProxyLabel(VertexLabel proxyLabel) {
+        if (this.identifyingNamespace != null) {
+            proxyLabel = proxyLabel.withoutOuterNamespace();
+        }
+        return proxyLabel.withoutOuterNamespace().withoutOuterNamespace();
+    }
```

Comment:
I'm not sure this is gonna be sufficient. Users are allowed to add their own namespaces too - although it's unlikely, it is allowed, and once we do more work on model composition it will become more common.

So you need to handle the case  `Sequence_Item_TIMESTEP.HASHCODE.proxy_for.SCOPED.LABEL`

In other words, your logic needs to be "drop the outer namespaces until you reach the `PROXY_LABEL_MARKER`"

Reply:
@migwellian I believe in this case the code will just return the label `proxy_for.SCOPED.LABEL`. I think this is actually what we want though.
Because then the `proxy_for.SCOPED.LABEL` will be default parent label as intended?

User defined scopes cant be added to form e.g. `Sequence_Item_TIMESTEP.HASHCODE.SCOPED.proxy_for.LABEL`, so I think this might still work.
Or have I just misunderstood your point?

Reply:
yes you're right. carry on.